sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/client/DefaultHttpClient.java

```diff
@@ -202,30 +205,53 @@ private void sendBody(HttpRequest httpRequest, HttpURLConnection connection) {
-        try {
-            int responseCode = connection.getResponseCode();
-            HttpHeaders responseHeaders = getResponseHeaders(connection);
+        HttpHeaders responseHeaders = getResponseHeaders(connection);
+        HttpResponse<?> httpResponse = createHttpResponse(httpRequest, connection);
 
+        if (connection.getErrorStream() == null && isTextEventStream(responseHeaders)) {
             ServerSentEventListener listener = httpRequest.getServerSentEventListener();
-            if (connection.getErrorStream() == null && isTextEventStream(responseHeaders)) {
-                if (listener != null) {
-                    processTextEventStream(httpRequest, connection, listener);
-                } else {
-                    LOGGER.atInfo().log("No listener attached to the server sent "
-                        + "event http request. Treating response as regular response.");
-                }
 
-                return new HttpResponse<>(httpRequest, responseCode, responseHeaders, null);
+            if (listener != null) {
+                processTextEventStream(httpRequest, connection, listener);
             } else {
-                AccessibleByteArrayOutputStream outputStream = getAccessibleByteArrayOutputStream(connection);
-
-                return new HttpResponse<>(httpRequest, responseCode, responseHeaders,
-                    BinaryData.fromByteBuffer(outputStream.toByteBuffer()));
+                LOGGER.atInfo().log("No listener attached to the server sent event http request. Treating response as "
+                    + "regular response.");
             }
+        } else {
+            HttpRequestMetadata metadata = httpRequest.getMetadata();
+
+            if (metadata != null && metadata.getResponseBodyHandling() != null) {
+                switch (metadata.getResponseBodyHandling()) {
+                    case IGNORE:
+                        ignoreResponseBody(httpResponse, connection);
+
+                        break;
+                    case STREAM:
+                        setStreamingResponseBody(httpResponse, connection);
+
+                        break;
+                    case BUFFER:
+                    case DESERIALIZE:
+                    default:
+                        eagerlyBufferResponseBody(httpResponse, connection);
+                }
+            } else {
+                // The metadata can only be null if intentionally set through HttpRequest.setMetadata()
+                // TODO (vcolin7): Should we ensure the metadata is not null when users call HttpRequest.setMetadata()?
+                eagerlyBufferResponseBody(httpResponse, connection);
+            }
+        }
+
+        return httpResponse;
+    }
+
+    private HttpResponse<?> createHttpResponse(HttpRequest httpRequest, HttpURLConnection connection) {
+        try {
+            return new HttpResponse<>(httpRequest, connection.getResponseCode(), getResponseHeaders(connection), null);
```

Comment:
Would it make sense to remove the body from the constructor as with this PR it seems like we want it to be only a settable property.

Reply:
The value and the body are two different properties and we want to have suers be able to provide a value at creation time, but not the body.

Reply:
Not in this case. The `HttpResponse` constructor needs to be able to take a `value` because that's the default implementation we return when calling `Response.createResponse()`, which also takes a `vaule`.

Reply:
As a side note, `value` and `body` are two different properties in `HttpResponse`. The former is the deserialized contents (an instance of `T`) of the response coming from the wire and the latter is the raw contents (`BinaryData`). We want users to be able to provide a `value` but not a `body`, which is why there's no constructor overload or public setter that allows for passing the latter, making us rely on a helper to modify the private value when required.